federation/ldap/src/main/java/org/keycloak/storage/ldap/idm/store/ldap/LDAPContextManager.java

```diff
@@ -238,7 +207,6 @@ public static Hashtable<Object, Object> getNonAuthConnectionProperties(LDAPConfi
 
     @Override
     public void close() {
-        if (vaultStringSecret != null) vaultStringSecret.close();
```

Comment:
I'm suggesting not having `vaultStringSecret` as a member variable all, since `Optional` complicated finding the bind password unnecessarily.  I'm suggesting having new method that handles that in single place:

```java
    // Get bind password from vault or from directly from configuration, may be null.
    private String getBindPassword() {
        VaultStringSecret vaultSecret = session.vault().getStringSecret(ldapConfig.getBindCredential());
        return vaultSecret.get().orElse(ldapConfig.getBindCredential());
    }
```